.circleci/config.yml

```diff
@@ -9,80 +9,76 @@ experimental:
             type: string
             description: relative path where the build must be run
             default: casskop
-        executor: vm-classic
+        executor: vm
         working_directory: /home/circleci/<< parameters.operatorDir >>
-        # ENVIRONMENTS VARIABLES
         environment:
-          K8S_VERSION: v1.14.0
-          KUBECONFIG: /home/circleci/.kube/config
-          MINIKUBE_VERSION: v1.0.0
-          MINIKUBE_WANTUPDATENOTIFICATION: false
-          MINIKUBE_WANTREPORTERRORPROMPT: false
-          MINIKUBE_HOME: /home/circleci
-          MINIKUBE_CONFIG: /home/circleci/.minikube
-          CHANGE_MINIKUBE_NONE_USER: true
+          K3S_IMAGE: rancher/k3s:v1.16.10-k3s1
+          K8S_VERSION: v1.17.5
           GOPATH: /go
           GO111MODULE: on
           DOCKER_LATEST: 1
           GIN_MODE: release
-        # STEPS
         steps:
-          # Pre-checkout steps
-          - run: # Install Minikube components to be able to run it.
-              name: Setup minikube
-              command: |
-                wget -O minikube https://github.com/kubernetes/minikube/releases/download/${MINIKUBE_VERSION}/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
-          - run: # Run minikube with expected parameters
-              name: Start minikube
-              background: true
-              command: |
-                sudo -E minikube start --vm-driver=none --cpus 4 --memory 4096 --kubernetes-version=${K8S_VERSION}
-          - run: # Install kubectl cli
-              name: Setup kubectl
+          - run:
+              name: Set up K3d
               command: |
-                wget -O kubectl https://storage.googleapis.com/kubernetes-release/release/${K8S_VERSION}/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+                wget -q -O - https://raw.githubusercontent.com/rancher/k3d/master/install.sh | bash
+                k3d create --image ${K3S_IMAGE} --wait 0
                 mkdir -p ${HOME}/.kube
-                touch ${HOME}/.kube/config
-          # Checkout steps
+                cp $(k3d get-kubeconfig) ${HOME}/.kube/config
           - checkout:
               path: /home/circleci/casskop
+          - run:
+              name: Create file with build image version
+              command: |
+                make build-image > casskop-build-image
+          - restore_cache:
+              keys:
+                - '{{ checksum "casskop-build-image" }}'
+          - run:
+              name: Load Casskop build image into k3d
+              command: |
+                if [ -f "casskop-build-image.tgz" ]
+                then
+                  docker load < casskop-build-image.tgz
+                else
+                  docker pull $(cat casskop-build-image)
+                  docker save $(cat casskop-build-image)|gzip > casskop-build-image.tgz
+                fi
+                k3d import-images $(cat casskop-build-image)
```

Comment:
why do you need to load this image in minikube, and we did'nt have to on minikube ? What is the difference ?

Reply:
I cache the image instead of pulling it ever time a test start. It should be faster than pulling 1GB I suppose